provider/provider.go

```diff
@@ -96,12 +96,16 @@ func (p *Provider) GetProviderConfig(_ context.Context, _ *cqproto.GetProviderCo
 
 func (p *Provider) ConfigureProvider(_ context.Context, request *cqproto.ConfigureProviderRequest) (*cqproto.ConfigureProviderResponse, error) {
```

Comment:
`error` is unused in the signature but removing it fails the CQServer interface check (which is unrelated I think...)

Reply:
We should usually avoid returning error and put it in the response, as that causes the GRPC fatal which no user understands why its "fatal"